conformance_tests/core/test_cmdqueue/src/test_cmdqueue.cpp

```diff
@@ -235,6 +235,155 @@ TEST_P(
+                                    &command_list_regular));
+      EXPECT_NE(nullptr, command_list_regular);
+
+      // Copy from host-allocated to device-allocated memory
+      EXPECT_EQ(ZE_RESULT_SUCCESS,
+                zeCommandListAppendMemoryCopy(
+                    command_list_regular, deive_buffer, host_buffer,
+                    buff_size_bytes, nullptr, 0, nullptr));
+
+      EXPECT_EQ(ZE_RESULT_SUCCESS,
+                zeCommandListAppendBarrier(command_list_regular, nullptr, 0,
+                                           nullptr));
+
+      // Copy from device-allocated memory back to host-allocated memory
+      EXPECT_EQ(ZE_RESULT_SUCCESS,
+                zeCommandListAppendMemoryCopy(command_list_regular, host_buffer,
+                                              deive_buffer, buff_size_bytes,
+                                              nullptr, 0, nullptr));
+
+      EXPECT_EQ(ZE_RESULT_SUCCESS, zeCommandListClose(command_list_regular));
+      list_of_command_lists.push_back(command_list_regular);
+    }
+  }
+
+  void TearDown() override {
+    for (uint32_t i = 0; i < params.num_command_lists; i++) {
+
+      EXPECT_EQ(
+          0, memcmp(host_buffers.at(i), verification_buffer, buff_size_bytes));
+
+      EXPECT_EQ(ZE_RESULT_SUCCESS,
+                zeCommandListDestroy(list_of_command_lists.at(i)));
+      lzt::free_memory(host_buffers.at(i));
+      lzt::free_memory(device_buffers.at(i));
+    }
+    lzt::destroy_command_queue(command_queue);
+  }
+
+  static const uint32_t buff_size_bytes = 12;
+  uint8_t verification_buffer[buff_size_bytes];
+  CustomExecuteParams params = GetParam();
+
+  ze_command_queue_handle_t command_queue = nullptr;
+  std::vector<ze_command_list_handle_t> list_of_command_lists;
+
+  std::vector<uint8_t *> host_buffers;
+  std::vector<uint8_t *> device_buffers;
+
+  ze_command_list_handle_t command_list_immediate;
+};
+
+class zeCommandListImmediateAppendCommandListsExpTestsHostSynchronize
+    : public zeImmediateCommandListAppendCommandListsExpTests {};
+
+TEST_P(
+    zeCommandListImmediateAppendCommandListsExpTestsHostSynchronize,
+    GivenCommandListImmediateAppendCommandListsExpAndSyncUsingCommandListHostSynchronizeThenCallSucceeds) {
+
+  auto command_lists_initial = list_of_command_lists;
+  EXPECT_EQ(ZE_RESULT_SUCCESS,
+            zeCommandListImmediateAppendCommandListsExp(
+                command_list_immediate, params.num_command_lists,
+                list_of_command_lists.data(), nullptr, 0, nullptr));
+
+  for (int i = 0; i < list_of_command_lists.size(); i++) {
+    ASSERT_EQ(list_of_command_lists[i], command_lists_initial[i]);
+  }
+
+  ze_result_t sync_status = ZE_RESULT_NOT_READY;
+  while (sync_status != ZE_RESULT_SUCCESS) {
+    EXPECT_EQ(sync_status, ZE_RESULT_NOT_READY);
+    sync_status = zeCommandListHostSynchronize(command_list_immediate,
+                                               params.sync_timeout);
+    std::this_thread::yield();
```

Comment:
This probably isn't needed [ref](https://www.realworldtech.com/forum/?threadid=189711&curpostid=189752)

Reply:
Because we are a single process/thread app (this ULT)?

Reply:
The reference I've posted above shows the reason why `yield` shouldn't be used even in multi threaded applications, then for single-threaded application it's not needed at all